### PR TITLE
Am feature profile

### PR DESCRIPTION
**Pull request by amarquesmoura**
this adds content to Profile page

### CODE REVIEW

File: client/js/navigation/config.js [hunk 1]
```diff
@@ -23,7 +26,30 @@ export const sharedNavigationOptions = navigation => ({
   },
   headerLeft: (
     <TouchableOpacity onPress={() => navigation.toggleDrawer()}>
-      <Icon name="md-menu" size={25} color="white" style={{ margin: 15 }} />
+      <IconEntypo
+        name="menu"
+        size={45}
+        color="#fff"
+        style={{ marginLeft: 10 }}
```
Comment (teelau) [inline client/js/navigation/config.js]: Move this to a style object

File: client/js/navigation/config.js [hunk 2]
```diff
@@ -23,7 +26,30 @@ export const sharedNavigationOptions = navigation => ({
...
+      />
+    </TouchableOpacity>
+  ),
+  headerTitle: (
+    <TouchableOpacity>
+      <Image
+        source={require("../assets/images/bikelockerlogo-white.png")}
+        style={{ width: 40, height: 40, marginBottom: 5 }}
```
Comment (teelau) [inline client/js/navigation/config.js]: Move to a style object

File: client/js/navigation/config.js [hunk 3]
```diff
@@ -23,7 +26,30 @@ export const sharedNavigationOptions = navigation => ({
...
+      />
+    </TouchableOpacity>
+  ),
+  headerRight: (
+    <TouchableOpacity>
+      <IconFontAwesome
+        name="location-arrow"
+        size={35}
+        color="#fff"
+        style={{ marginRight: 15, marginBottom: 5 }}
```
Comment (teelau) [inline client/js/navigation/config.js]: Move to a style object

File: client/js/screens/Profile/Profile.js [hunk 4]
```diff
@@ -1,10 +1,36 @@
...
 
-const Profile = ({ params }) => (
-  <View>
-    <Text>This is the Profile Screen</Text>
-  </View>
-);
+const Profile = ({ user }) => {
+  console.log("user", user);
+  return (
+    <ScrollView>
```
Comment (teelau) [inline client/js/screens/Profile/Profile.js]: Is ScrollView required?

File: client/js/screens/Profile/Profile.js [hunk 5]
```diff
@@ -1,10 +1,36 @@
...
+      <View>
+        <Text>My Profile</Text>
+        <TouchableOpacity>
+          <Text>Edit</Text>
+        </TouchableOpacity>
+      </View>
+      <View>
+        <Image
+          style={{ width: 200, height: 200, borderRadius: 100 }}
```
Comment (teelau) [inline client/js/screens/Profile/Profile.js]: Move this to a style object

File: client/js/context/UserProvider.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  );
+};
+
+ViewerProvider.propTypes = {
+  children: PropTypes.object.isRequired
+};
+
+const USER_QUERY = gql`
+  query User($id: ID) {
+    User(id: "cjwtisae10ech0103e84u962r") {
```
Comment (teelau) [inline client/js/context/UserProvider.js]: Is this a `secret`? Move to a .env and use react-native-dotenv. Squash the commit so this key is not available publicly

**Review by teelau — CHANGES_REQUESTED** [review]

File: client/js/navigation/config.js (hunk 1 above)
Comment (amarquesmoura) [reply client/js/navigation/config.js]: This was part of the navigation feature.

File: client/js/navigation/config.js (hunk 2 above)
Comment (amarquesmoura) [reply client/js/navigation/config.js]: This was part of the navigation feature.

File: client/js/navigation/config.js (hunk 3 above)
Comment (amarquesmoura) [reply client/js/navigation/config.js]: This was part of the navigation feature.

File: client/js/screens/Profile/Profile.js (hunk 5 above)
Comment (amarquesmoura) [reply client/js/screens/Profile/Profile.js]: Will move it to the style.js in the next step

File: client/js/context/UserProvider.js (hunk 6 above)
Comment (amarquesmoura) [reply client/js/context/UserProvider.js]: This is the id of a sample user. It will be substituted by the current user id once context is implemented.

**Comment by amarquesmoura** [comment]
Done.

File: client/js/screens/Profile/Profile.js (hunk 4 above)
Comment (amarquesmoura) [reply client/js/screens/Profile/Profile.js]: I have changed it to View

File: client/js/screens/Profile/styles.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    color: theme.black,
+    fontWeight: "bold"
+  },
+  link: {
+    ...theme.link
+  },
+  profilePic: {
+    width: 150,
+    height: 150,
+    borderRadius: 75
```
Comment (siddharthparmar7) [inline client/js/screens/Profile/styles.js]: Please use flex with `resizeMode: "contain"` instead of fixed height and width.

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Nice work! Please make the changes and resolve merge conflicts on this branch. Thank you.

**Comment by amarquesmoura** [comment]
Moved gql query to ProfileContainer and removed context.

File: client/js/screens/Profile/Profile.js [hunk 8]
```diff
@@ -1,10 +1,37 @@
...
+import { Image, Text, TouchableOpacity, ScrollView, View } from "react-native";
+import styles from "./styles";
 
-const Profile = ({ params }) => (
-  <View>
-    <Text>This is the Profile Screen</Text>
-  </View>
-);
+const Profile = ({ user }) => {
+  console.log("user", user);
```
Comment (teelau) [inline client/js/screens/Profile/Profile.js]: remove console log

**Review by teelau — CHANGES_REQUESTED** [review]

File: client/js/screens/Profile/Profile.js (hunk 8 above)
Comment (amarquesmoura) [reply client/js/screens/Profile/Profile.js]: Done.

File: client/js/screens/Profile/styles.js (hunk 7 above)
Comment (amarquesmoura) [reply client/js/screens/Profile/styles.js]: Issue solved.

File: client/js/screens/Profile/ProfileContainer.js [hunk 9]
```diff
@@ -8,6 +11,42 @@ export default class ProfileContainer extends Component {
   }
```
Comment (siddharthparmar7) [inline client/js/screens/Profile/ProfileContainer.js]: Remove this contractor if you are not using a state.

File: client/js/screens/Profile/ProfileContainer.js [hunk 10]
```diff
@@ -8,6 +11,42 @@ export default class ProfileContainer extends Component {
...
   render() {
-    return <Profile />;
+    return (
+      <Query
+        query={USER_QUERY}
+        variables={{ id: "cjx2bg22z08h90166zqgy3ium" }}
+        fetchPolicy="network-only"
+      >
+        {({ loading, error, data }) => {
+          console.log("data", data);
```
Comment (siddharthparmar7) [inline client/js/screens/Profile/ProfileContainer.js]: Please remove this console log.

File: client/js/screens/Profile/ProfileContainer.js [hunk 11]
```diff
@@ -8,6 +11,42 @@ export default class ProfileContainer extends Component {
...
+          if (loading) return <Text>Loading...</Text>; // replace by Loader component
```
Comment (siddharthparmar7) [inline client/js/screens/Profile/ProfileContainer.js]: Please use the Loading screen instead.

File: client/js/screens/Profile/ProfileContainer.js [hunk 12]
```diff
@@ -8,6 +11,42 @@ export default class ProfileContainer extends Component {
...
+          if (error) return <p>{`Error! ${error.message}`}</p>;
```
Comment (siddharthparmar7) [inline client/js/screens/Profile/ProfileContainer.js]: React Native doesn't have a `p` tag. Use React Native components.

File: client/js/screens/Profile/styles.js [hunk 13]
```diff
@@ -0,0 +1,55 @@
+import theme from "../../config/globalStyles";
+
+const styles = {
+  container: {
+    ...theme.container,
+    height: "100%",
+    marginLeft: "7%",
+    marginRight: "7%",
```
Comment (siddharthparmar7) [inline client/js/screens/Profile/styles.js]: Please use `marginHorizontal: (int value)`. Thank you.

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Nice work! Please make the changes. Thank you.

File: client/js/screens/Profile/ProfileContainer.js (hunk 9 above)
Comment (amarquesmoura) [reply client/js/screens/Profile/ProfileContainer.js]: Done.

File: client/js/screens/Profile/ProfileContainer.js (hunk 10 above)
Comment (amarquesmoura) [reply client/js/screens/Profile/ProfileContainer.js]: Done.

File: client/js/screens/Profile/ProfileContainer.js (hunk 11 above)
Comment (amarquesmoura) [reply client/js/screens/Profile/ProfileContainer.js]: Done.

File: client/js/screens/Profile/ProfileContainer.js (hunk 12 above)
Comment (amarquesmoura) [reply client/js/screens/Profile/ProfileContainer.js]: Done.

File: client/js/screens/Profile/styles.js (hunk 13 above)
Comment (amarquesmoura) [reply client/js/screens/Profile/styles.js]: Done.

File: client/js/screens/Profile/Profile.js [hunk 14]
```diff
@@ -1,10 +1,39 @@
...
+        <Text style={styles.caption}>
+          {user.bikeColor} {user.bikeBrand} {user.bikeType}
+        </Text>
+      </View>
+    </View>
+  );
+};
 
-export default Profile;
+export default withNavigation(Profile);
```
Comment (siddharthparmar7) [inline client/js/screens/Profile/Profile.js]: Pass the navigation prop from the screen container instead of using `withNavigation`. Thank you.

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]

File: client/js/screens/Profile/Profile.js (hunk 14 above)
Comment (amarquesmoura) [reply client/js/screens/Profile/Profile.js]: Done.

**Review by siddharthparmar7 — APPROVED** [review]

**Pull request merged**